test/sql/aggregate/aggregates/test_list_aggregate_function.test

```diff
@@ -0,0 +1,187 @@
+# name: test/sql/aggregate/aggregates/test_list_aggregate_function.test
+# description: Test the list aggregate function for all data types
+# group: [aggregates]
+
+# INTEGER types
+
+foreach type tinyint smallint integer bigint utinyint usmallint uinteger ubigint
+
+statement ok
+CREATE TABLE test (g INTEGER, i ${type})
+
+statement ok
+INSERT INTO test VALUES (1, 2), (1, 3), (2, 4)
+
+query I
+SELECT LIST(i) FROM test GROUP BY g ORDER BY g
+----
+[2, 3]
+[4]
+
+statement ok
+DROP TABLE test
+
+endloop
+
+# FLOAT, DOUBLE
+
+foreach type float double
+
+statement ok
+CREATE TABLE test (g INTEGER, i ${type})
+
+statement ok
+INSERT INTO test VALUES (1, 2), (1, 3), (2, 4)
+
+query I
+SELECT LIST(i) FROM test GROUP BY g ORDER BY g
+----
+[2.0, 3.0]
+[4.0]
+
+statement ok
+DROP TABLE test
+
+endloop
+
+# VARCHAR
+
+statement ok
+CREATE TABLE varch (g integer, str varchar)
+
+statement ok
+INSERT INTO varch VALUES (1, 'hello'), (1, 'was'), (2, 'geht')
```

Comment:
Can we get a test with really long strings? Like strings that are longer than the default allocator size.

You can create really long strings with `select repeat('a', 10000);`